### PR TITLE
256modules

### PR DESCRIPTION
**Pull request by stuartpittaway**

### CODE REVIEW

**Comment by daromer2** [comment]
I guess that PWM could affect the communication due to the frequency causing induced errors? Having the ON vs 100% PWM should only reduce it somewhat since with current layout the duty cycle look to be less than 50% due to heat dissipation. Ie what i can see it runs PWM most of the time. I havent been able to track any comm errors based on PWM as of now or atleast havent looked at it. Though i have some other interesting comm errors that i need to dig into. 
In short: Im to new into this code to be able to answer if using only PWM is ok or not.  I guess code would be simpler with only PWM but i need to get my larger test system up and running before i feel i can have enough test-data. 

If you want to add the code calculating mAh on cell module level we could skip part of above code.

**Comment by daromer2** [comment]
Lets discard this pull request.
